tests/purs/failing/MissingRecordField.out

```diff
@@ -2,7 +2,20 @@ Error found:
 in module [33mMissingRecordField[0m
 at tests/purs/failing/MissingRecordField.purs:10:19 - 10:23 (line 10, column 19 - line 10, column 23)
 
-  Type of expression lacks required label [33mage[0m.
```

Comment:
This is something I mentioned earlier: the new error is less clear about which side is expected and which side is actual. This'll come up again in the `checkProperties` case.

I think we should add a boolean to `TypesDoNotUnify` to flag whether the types in question are known to be in actual-expected order, and give a different error message if true (`The type * does not match the expected type *`, perhaps?)